### PR TITLE
Fix error returned when PURCHASE_EXPIRED

### PR DESCRIPTION
**Pull request by rastafan**
Fixing what i think is a mistake in the returned error code when validator returns PURCHASE_EXPIRED.

Please check if i understood the logic right.

### CODE REVIEW

File: src/js/product.js [hunk 1]
```diff
@@ -281,7 +281,7 @@ store.Product.prototype.verify = function() {
                     store.log.debug("verify -> server did use the latest_receipt, no retries");
                     nRetry = 999999;
                 }
-                if (data.code === store.PURCHASE_EXPIRED) {
+                if (data.code === store.ERR_PAYMENT_EXPIRED) {
```
Comment (j3k0) [inline src/js/product.js:284]: This change is not correct.

Receipt validator errors have codes `6778*`.

Error with codes `6777*`, named `ERR_`, are triggered by the plugin.

See `constants.js` for the list of validator error codes.

File: src/js/product.js [hunk 2]
```diff
@@ -295,6 +295,10 @@ store.Product.prototype.verify = function() {
                         });
                     }
                     else {
+                        err = new store.Error({
+                            code: store.PURCHASE_EXPIRED,
```
Comment (j3k0) [inline src/js/product.js:299]: Cf comment above, a valid error code should be `ERR_*` (like `ERR_PAYMENT_EXPIRED`).

**Comment by j3k0** [comment]
Sorry, can't accept this PR (breaks the existing behavior with non-needed change to error codes -- also making the plugin trigger error codes reserved to the validator)

**Comment by rastafan** [comment]
I guess this makes sense, thanks for taking the time to check on this.


Also, may i ask if the PURCHASE_EXPIRED error is intended for the payment (a transaction left incomplete for too long) or for the subscription (a subscription whose time has ended and has not been renewed)?

Also, since this triggers both the "error" and "expired" callbacks, should we check on the error code in the two and fire an error message from one or the other based on the code returned?

**Pull request closed without merging**